src/semantic/mod.rs

```diff
@@ -28,6 +28,19 @@ impl SemanticChecker {
                 Class(c) => {
                     let typ = self.type_env.new_class(&top.tag, c)?;
                     self.type_env.add_type(&c.location, &c.name, typ)?;
+                }
+                _ => (),
+            }
+        }
+        for top in ast {
+            use TopAstVariant::*;
+            match &top.ast {
+                Class(c) => {
+                    let class_type = self.type_env.get_type(&c.location, &c.name)?;
+                    match class_type.typ {
+                        types::Type::ClassType { .. } => {}
+                        _ => (),
+                    }
```

Comment:
Here the target is to make sure each type already been emitted before others type uses it. But seems like 40 to 43 is useless code.

Reply:
I already remove the unused code but leave the comment for your review.